### PR TITLE
Make file attachments modal

### PR DESCRIPTION
**Pull request by balintbartok**
refs: MBL-14740
affects: Student, Teacher
release note: Fixed file preview window in some cases

### CODE REVIEW

File: Core/Core/Discussions/DiscussionReplyViewController.swift [hunk 1]
```diff
@@ -329,3 +329,14 @@ extension DiscussionReplyViewController: FilePickerDelegate, QLPreviewController
         return attachmentURL! as NSURL
     }
 }
+
+extension DiscussionReplyViewController: CoreWebViewLinkDelegate {
```
Comment (vanwagonet) [inline Core/Core/Discussions/DiscussionReplyViewController.swift:333]: nice catch remembering the reply also!

File: Core/Core/Discussions/DiscussionDetailsViewController.swift [hunk 2]
```diff
@@ -450,7 +450,11 @@ extension DiscussionDetailsViewController: CoreWebViewLinkDelegate {
             url.host == env.currentSession?.baseURL.host,
             url.path.hasPrefix("/\(context.pathComponent)/discussion_topics/\(topicID)/")
         else {
-            env.router.route(to: url, from: self)
+            if url.pathComponents.count > 1, url.pathComponents[1] == "files" {
```
Comment (vanwagonet) [inline Core/Core/Discussions/DiscussionDetailsViewController.swift:453]: Do we also get links to files with a course or user context sometimes?

**Review by vanwagonet — COMMENTED** [review]
I'll test when the build is ready, but we might need to also check for course & user files, not just urls to files that omit the context.

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/51292805/p/4a4dda49d5085264b7eff88ea1997ff3">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/51292823/p/9e721f75695c92000aeb3ccd291d62e7">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/51292758/p/389c36c9228512dc8ac5d2d32c13231b">
    
  </a>
</details>

**Review by vanwagonet — APPROVED** [review]
QA+1 attachments are consistently opening in a modal for me.

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.93% | 92.93% | 0%

[MBL-14740](https://instructure.atlassian.net/browse/MBL-14740)

#### Affected Apps: Student, Teacher

#### Release Note: 
Fixed file preview window in some cases
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**